modules/learn/pages/clusters-and-availability/automatic-failover.adoc

```diff
@@ -8,39 +9,72 @@
+
+In the xref:learn:clusters-and-availability/automatic-failover.adoc#failover-events[Failover Events] section, the events that start the auto-failover process are described.  In the xref:learn:clusters-and-availability/automatic-failover.adoc#auto-failover-constraints[Auto-Failover Constraints] section, the safety checks done after an auto-failover triggering event are explained.  These safety checks are done to determine if an auto-failover should actually be performed. Checks required for service specific safety policies are described in the xref:learn:clusters-and-availability/automatic-failover.adoc#failover-policy[Service-Specific Auto-Failover Policy] section.
+
 For information on managing auto-failover, see the information provided for Couchbase Web Console at xref:manage:manage-settings/general-settings.adoc[General] (which provides information on general cluster-settings), for the REST API at xref:rest-api:rest-cluster-autofailover-intro.adoc[Managing Auto-Failover], and for the CLI at xref:cli:cbcli/couchbase-cli-setting-autofailover.adoc[setting-autofailover].
 
 == Failover Events
 
-Auto-failover occurs in response to failed/failing events.
-There are three types of event that can trigger auto-failover:
+Auto-failover occurs in response to failed/failing events. Auto-failover applies to the node -- it’s the node that fails over regardless of the triggering event. There are specific types of events that trigger auto-failover processing. However, auto-failover will only actually occur if all of the checks (constraints and policies) for auto-failover pass.  
+
+The constraints and policies checked after the triggering event are described in xref:learn:clusters-and-availability/automatic-failover.adoc#auto-failover-constraints[Auto-Failover Constraints] and xref:learn:clusters-and-availability/automatic-failover.adoc#failover-policy[Service-Specific Auto-Failover Policy] sections. For example, despite the triggering event, an auto-failover of a node may be prevented due to the lack of a quorum or because there aren’t any other nodes running a service that exists on the node to be auto-failed over.
+
+Below are the types of events that can trigger auto-failover processing on a node:
 
 * _Node failure_.
-A server-node within the cluster is unresponsive (due to a network failure, out-of-memory problem, or other node-specific issue).
+A server-node within the cluster is unresponsive (due to a network failure, very high CPU utilization problem, out-of-memory problem, or other node-specific issue). This means that the ns-server process on the node has not responded to the cluster manager for the user-specified configured amount of time -- the health of the services running on the node is unknown.
```

Comment:
Similarly to the previous comment, we might want to revise mention of the ns-server process and just talk about the cluster manager heatbeat

Reply:
With Ben's suggested change, line 42 may be changed to:

A server-node within the cluster is unresponsive (due to a network failure, very high CPU utilization problem, out-of-memory problem, or other node-specific issue). This means that the the cluster manager of the node has not sent heartbeats in the configured timeout period, and therefore, the health of the services running on the node is unknown.

Note for Supritha:  The "Node-Failure Detection" [section](https://docs.couchbase.com/server/current/learn/clusters-and-availability/cluster-manager.html) of Cluster Manager doc has info on the heartbeat mechanism in case some context is needed.